src/ethz_snow/snowing.py

```diff
@@ -0,0 +1,2227 @@
+            )
+        else:
+            self.k = k
+
+        # for saving the results
+        self._domain = None
+        self._prop = None
+        self._time = None
+        self._shelf = None
+        self._temp = None
+        self._ice = None
+
+        # dictionary of results
+        self._stats = {}
+
+        # number of repetitions
+        self.Nrep = Nrep
+
+        # dictionary of results for multiple simulations
+        if self.Nrep > 1:
+            self._statsMultiple = {}
+
+        # path to the parameters file
+        self.configPath = configPath
+
+        # simulation progress
+        self._simulationStatus = 0
+
+        # model complexity
+        if temperature is not None:
+            self.temperature = temperature
+            self.const["temperature"] = temperature
+        else:
+            self.temperature = self.const["temperature"]
+
+        # freezing configuration (shelf, VISF, etc.)
+        if configuration is not None:
+            self.configuration = configuration
+            self.const["configuration"] = configuration
+        else:
+            self.configuration = self.const["configuration"]
+
+        # check configuration
+        if self.configuration not in ["shelf", "VISF", "jacket"]:
+            raise NotImplementedError(
+                (
+                    f'Configuration "{self.configuration}" '
+                    + "not correctly specified, use shelf, jacket or VISF."
+                )
+            )
+
+        # check complexity
+        if self.temperature not in ["homogeneous", "spatial_1D", "spatial_2D"]:
+            raise NotImplementedError(
+                (
+                    f'Model complexity "{self.temperature}" '
+                    + "not correctly specified, use homogeneous, spatial_1D or spatial_2D."
+                )
+            )
+
+        # plotting
+        if plotting is not None:
+            self.plotting = plotting
+        else:
+            self.plotting = True
+
+        # oeprating conditions
+        if not isinstance(opcond, OperatingConditions):
+            raise TypeError(
+                "Input opcond must be an instance of class OperatingConditions."
+            )
+        else:
+            self.opcond = opcond
+
+    @property
+    def configPath(self) -> Optional[str]:
+        """Get configPath property."""
+        return self._configPath
+
+    @configPath.setter
+    def configPath(self, value):
+        """Set configPath property."""
+        self.const = calculateDerived(value)
+        self._configPath = value
+
+    @property
+    def simulationStatus(self) -> int:
+        """Return simulation status of instance.
+
+        Returns:
+            int: Simulation status. 0 = not run, 1 = run.
+        """
+        if (self._simulationStatus == 1) or (self._temp is not None):
+            self._simulationStatus = 1
+
+        return self._simulationStatus
+
+    @property
+    def getTime(self) -> np.ndarray:
```

Comment:
Here, `getTime` is a property and the `@property` decorator defines this to be a getter function. A more natural name for this variable would be "time"

Reply:
The same holds true for all the other `get...` properties, of course

Reply:
Makes sense, changed.